kafka/src/main/java/com/metamx/tranquility/kafka/writer/WriterController.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.kafka.writer;
+
+import com.metamx.common.logger.Logger;
+import com.metamx.tranquility.config.DataSourceConfig;
+import com.metamx.tranquility.kafka.model.MessageCounters;
+import com.metamx.tranquility.kafka.model.TranquilityKafkaConfig;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+/**
+ * Manages the creation and operation of TranquilityEventWriters.
+ */
+public class WriterController
+{
+  private static final Logger log = new Logger(WriterController.class);
+
+  private List<DataSourceConfig<TranquilityKafkaConfig>> dataSourceConfigList;
+  private Map<String, TranquilityEventWriter> writers = new ConcurrentHashMap<>();
+
+  public WriterController(Map<String, DataSourceConfig<TranquilityKafkaConfig>> dataSourceConfigs)
+  {
+    this.dataSourceConfigList = new ArrayList<>(dataSourceConfigs.values());
+    this.dataSourceConfigList.sort(
+        new Comparator<DataSourceConfig<TranquilityKafkaConfig>>()
+        {
+          @Override
+          public int compare(DataSourceConfig<TranquilityKafkaConfig> o1, DataSourceConfig<TranquilityKafkaConfig> o2)
+          {
+            return o2.config().getTopicPatternPriority().compareTo(o1.config().getTopicPatternPriority());
+          }
+        }
+    );
+
+    log.info("Ready: [topicPattern] -> dataSource mappings (priority):");
```

Comment:
the `(priority)` is probably not needed as it's explicitly mentioned in the following log messages.